staging/src/k8s.io/apiserver/pkg/storage/preconditions_test.go

```diff
@@ -0,0 +1,192 @@
+			objExists:     true,
+		},
+		"UID is met": {
+			preconditions: NewUIDPreconditions(uid1),
+			objName:       "foo",
+			obj:           &example.Pod{ObjectMeta: metav1.ObjectMeta{Name: "foo", UID: types.UID(uid1)}},
+			objExists:     true,
+		},
+		"ResourceVersion is met": {
+			preconditions: NewResourceVersionPreconditions(resourceVersion1),
+			objName:       "foo",
+			obj:           &example.Pod{ObjectMeta: metav1.ObjectMeta{Name: "foo", ResourceVersion: resourceVersion1}},
+			objExists:     true,
+		},
+		"UID is met, ResourceVersion is not": {
+			preconditions: NewPreconditions(uid1, resourceVersion1),
+			objName:       "foo",
+			obj:           &example.Pod{ObjectMeta: metav1.ObjectMeta{Name: "foo", UID: types.UID(uid1), ResourceVersion: resourceVersion2}},
+			objExists:     true,
+			expectedErrorMessageWords: [][]string{
+				{"Precondition", "ResourceVersion", resourceVersion1, resourceVersion2},
+			},
+		},
+		"ResourceVersion is met, UID is not": {
+			preconditions: NewPreconditions(uid1, resourceVersion1),
+			objName:       "foo",
+			obj:           &example.Pod{ObjectMeta: metav1.ObjectMeta{Name: "foo", UID: types.UID(uid2), ResourceVersion: resourceVersion1}},
+			objExists:     true,
+			expectedErrorMessageWords: [][]string{
+				{"Precondition", "UID", uid1, uid2},
+			},
+		},
+		"neither UID nor ResourceVersion is met": {
+			preconditions: NewPreconditions(uid1, resourceVersion1),
+			objName:       "foo",
+			obj:           &example.Pod{ObjectMeta: metav1.ObjectMeta{Name: "foo", UID: types.UID(uid2), ResourceVersion: resourceVersion2}},
+			objExists:     true,
+			expectedErrorMessageWords: [][]string{
+				{"Precondition", "UID", uid1, uid2},
+				{"Precondition", "ResourceVersion", resourceVersion1, resourceVersion2},
+			},
+		},
+		"UID is not met because object doesn't exist": {
+			preconditions: NewUIDPreconditions(uid1),
+			obj:           &example.Pod{},
+			expectedErrorMessageWords: [][]string{
+				{"Precondition", "UID", "object", "exist"},
+			},
+		},
+		"ResourceVersion is not met because object doesn't exist": {
+			preconditions: NewResourceVersionPreconditions(resourceVersion1),
+			obj:           &example.Pod{},
+			expectedErrorMessageWords: [][]string{
+				{"Precondition", "ResourceVersion", "object", "exist"},
+			},
+		},
+		"neither UID nor ResourceVersion are met because object doesn't exist": {
+			preconditions: NewPreconditions(uid1, resourceVersion1),
+			obj:           &example.Pod{},
+			expectedErrorMessageWords: [][]string{
+				{"Precondition", "UID", "object", "exist"},
+				{"Precondition", "ResourceVersion", "object", "exist"},
+			},
+		},
+	}
+
+	for description, tc := range testCases {
+		t.Run(description, func(t *testing.T) {
+			gotErr := tc.preconditions.SafeCheck(tc.objName, tc.obj, tc.objExists)
+			if err := pass(gotErr, tc.expectedErrorMessageWords); err != nil {
+				t.Fatal(err)
+			}
```

Comment:
Can you also run the regular check method and verify that it returns the same pass / fail value (but no need to make assertions about the error it returns)? That way it will be harder for people to not update both functions.